File: scripts/UI.py
#!/usr/bin/env python3

import rospy
from turtlesim.msg import Pose
from geometry_msgs.msg import Twist
from turtlesim.srv import Spawn

turtle_name = "turtle1"
pub = rospy.Publisher(turtle_name + "/cmd_vel", Twist, queue_size=1)

Vx, Vy, Wz = 0,0,0

def turtleCallback(msg):
	global Vx, Vy, Wz
	#rospy.loginfo("Moving turtle pose @[%f, %f, %f]", msg.x, msg.y, msg.theta)
	
	vel = Twist()
	vel.linear.x = Vx
	vel.linear.y = Vy
	vel.angular.z = Wz
	
	print("Vx Vy Wz: ", Vx, Vy, Wz)
	pub.publish(vel)
	
	

def turtleMove():
	rospy.init_node("turtle_move")
	rospy.wait_for_service("/spawn")
	client2 = rospy.ServiceProxy("/spawn", Spawn)
	client2(2.0, 1.0, 0.0, "turtle2")
	rate = rospy.Rate(0.1)
	while not rospy.is_shutdown():
		global Vx, Vy, Wz, turtle_name
		turtle_name = input("Which turtle do you want to move (turtle1 or turtle2):")
		Vx = float(input("Vx:"))
		Vy = float(input("Vy:"))
		Wz = float(input("Wz:"))
		
		rate.sleep()
		
		pub = rospy.Publisher(turtle_name + '/cmd_vel', Twist, queue_size=1)
		rospy.Subscriber(turtle_name + '/pose', Pose, turtleCallback)
		print("Onara1")

		#rospy.sleep(1) 
		rate.sleep()
		print("Onara2")
		#rospy.spin()


if __name__ == '__main__':
	turtleMove()

File: src/UI.cpp
#include "ros/ros.h"
#include "turtlesim/Pose.h"
#include "geometry_msgs/Twist.h"
#include "turtlesim/Spawn.h"
#include <iostream>
#include <string>
#include <vector>

using namespace std;

ros::Publisher pub;

vector<float> speed_input(3);

void turtleCallback(const turtlesim::Pose::ConstPtr& msg)
	{
	
	ROS_INFO("Moving turtle pose@[%f, %f, %f]",
	msg->x, msg->y, msg->theta);
	geometry_msgs::Twist my_vel;	
	
	my_vel.linear.x = speed_input[0];
	my_vel.linear.y = speed_input[1];
	my_vel.angular.z = speed_input[2];
	
	cout << "Speed input:";
	cout << speed_input[0] <<";";
	cout << speed_input[1]<<";";
	cout << speed_input[2] << endl;
	
	pub.publish(my_vel);
	
	}
	
	
int main (int argc, char **argv)
{

	ros::init(argc, argv, "turtlebot_subscriber");
	ros::NodeHandle nh;
	
	
	ros::ServiceClient client1 = nh.serviceClient<turtlesim::Spawn>("/spawn");
	turtlesim::Spawn srv1;
	srv1.request.x = 2.0;
	srv1.request.y = 1.0;
	srv1.request.theta = 0.0;
	srv1.request.name="turtle2";
	client1.waitForExistence();
	client1.call(srv1);
	
	
	while (ros::ok()){
	
		string turtle_nb;
		cout << "Which one do you want to move? turtle1 or turtle2:" << endl;
		cin >> turtle_nb;
		cout << "Choose the speed {Vx, Vy, Wz}:" << endl;
		cout << "Vx:" << endl;
		cin >> speed_input[0];
		cout << "Vy:" << endl;
		cin >> speed_input[1];
		cout << "Wz:" << endl;
		cin >> speed_input[2];
		
		pub = nh.advertise<geometry_msgs::Twist>(turtle_nb + "/cmd_vel",1);
		ros::Subscriber sub = nh.subscribe(turtle_nb + "/pose",1,turtleCallback);
		
		//geometry_msgs::Twist my_vel;
		//my_vel.linear.x = speed_input[0];
		//my_vel.linear.y = speed_input[1];
		//my_vel.angular.z = speed_input[2];
		
		//pub.publish(my_vel);
		
		sleep(1);
		ros::spinOnce();
	}
	return 0;

}
